src/modules/utils.rs

```diff
@@ -161,19 +161,19 @@ pub fn get_file_type() -> &'static str {
     }
 }
 
-pub async fn is_version_installed(version: &str, config: &Config) -> bool {
-    let downloads_dir = get_downloads_folder(config).await.unwrap();
-    let mut dir = tokio::fs::read_dir(&downloads_dir).await.unwrap();
+pub async fn is_version_installed(version: &str, config: &Config) -> Result<bool> {
```

Comment:
Is this even needed? sure it allows usage `?` instead of `unwrap` but it will require you to "unresult" the result when it's simply a yes or no question.

Reply:
@orhun this is the only question I have regarding this PR, overall looks good!

Reply:
I think it's necessary for error handling. Otherwise you need to use `unwrap` or `expect` in the function which means it might panic.

It's a question about whether you'd choose proper error handling or a clear abstraction I guess.

Reply:
🤔 this is a good point actually